File: website/ClientApp/app/entities/book-copy.ts
export class BookCopy {
    barCode: string;
    isLost?: boolean;
    isDamaged?: boolean;

    static fromObject(bookCopyObj: any): BookCopy {
        const bookCopy = new BookCopy(bookCopyObj.barCode);
        Object.assign(bookCopy, bookCopyObj);
        return bookCopy;
    }

    /**
     *
     */
    constructor(barCode: string) {
        this.barCode = barCode;
    }
}

File: HomeReadingLibrary.Domain/Entities/BookCopy.cs
using System;

namespace HomeReadingLibrary.Domain.Entities
{
  public class BookCopy
  {
    public string BarCode { get; set; }
    public bool? IsLost { get; set; }
    public DateTime? LostDate { get; set; }
    public bool? IsDamaged { get; set; }
    public DateTime? DamagedDate { get; set; }
    public string Comments { get; set; }
  }
}